lib/cosmos/config/config_parser.rb

```diff
@@ -180,35 +180,25 @@ def parse_file(filename,
                    remove_quotes = true,
                    &block)
       @filename = filename
-      file = nil
-      unparsed_data = nil
-      begin
-        # Create a temp file where we can write the ERB parsed output
-        file = Tempfile.new("parsed_#{File.basename(filename)}")
-        unparsed_data = File.read(@filename)
-        file.write(ERB.new(unparsed_data).result(binding))
-        file.rewind
 
-        size = file.stat.size.to_f
+      # Create a temp file where we write the ERB parsed output
+      file = create_parsed_output_file(filename)
+      size = file.stat.size.to_f
 
-        # Callbacks for beginning of parsing
-        @@message_callback.call("Parsing #{size} bytes of #{filename}") if @@message_callback
-        @@progress_callback.call(0.0) if @@progress_callback
+      # Callbacks for beginning of parsing
+      @@message_callback.call("Parsing #{size} bytes of #{filename}") if @@message_callback
+      @@progress_callback.call(0.0) if @@progress_callback
 
+      begin
         # Loop through each line of the data
         parse_loop(file,
                    yield_non_keyword_lines,
                    remove_quotes,
                    size,
                    PARSING_REGEX,
                    &block)
-      rescue Exception => e
-        debug_file = create_debug_output_file(filename, file, unparsed_data, e)
-        if debug_file
-          raise e, "#{e}\nDebug output in #{debug_file}", e.backtrace
-        else
-          raise e
-        end
+      rescue => e
```

Comment:
I prefer to rescue Exception so that we will catch EVERYTHING and be able to handle it.